pkg/chunk/azure/blob_storage_client.go

```diff
@@ -41,6 +55,7 @@ func (c *BlobStorageConfig) RegisterFlags(f *flag.FlagSet) {
 
 // RegisterFlagsWithPrefix adds the flags required to config this to the given FlagSet
 func (c *BlobStorageConfig) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
+	f.StringVar(&c.CloudEnvironment, prefix+"azure.cloud_environment", "AzureGlobal", "Environment of Azure Cloud. This value can be AzureGlobal, AzureChinaCloud, AzureGermanCloud, AzureUSGovernment.")
```

Comment:
We have a convention to use `_` in YAML and `-` in CLI flags. Here it should be `cloud-environment`, even if I would suggest to just call it `environment`.

Also if you have a constant for `AzureGlobal` and the other environments and you create a global slice variable with all supported environments, then here you could dinamically generate it using `strings.Join(supportedEnvironments, ",")`.

Reply:
done